src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -25,60 +25,41 @@
 //
 //------------------------------------------------------------------------------
 
+using Microsoft.IdentityModel.Logging;
 using System.Collections.Generic;
-using System.Diagnostics.Tracing;
-using System.Globalization;
+using System.ComponentModel;
 using System.Security.Claims;
 using System.Security.Cryptography.X509Certificates;
 using System.Text;
 using System.Text.RegularExpressions;
-using System.Xml;
-using Microsoft.IdentityModel.Logging;
 using System.Threading;
+using System.Xml;
 
 namespace System.IdentityModel.Tokens.Jwt
 {
     /// <summary>
-    /// A <see cref="SecurityTokenHandler"/> designed for creating and validating Json Web Tokens. See http://tools.ietf.org/html/draft-ietf-oauth-json-web-token-07.
+    /// A <see cref="SecurityTokenHandler"/> designed for creating and validating Json Web Tokens. See: http://tools.ietf.org/html/rfc7519 and http://www.rfc-editor.org/info/rfc7515
     /// </summary>
     public class JwtSecurityTokenHandler : SecurityTokenHandler, ISecurityTokenValidator
     {
         private delegate bool CertMatcher(X509Certificate2 cert);
 
-        // the Sts pipeline expects the first identifier to be a string that 
-        // Uri.TryCreate( tokenIdentifiers[0], UriKind.Absolute, out result ) will be true.
-        // if that is not true, sts's using the .Net sts class will start failing.
-
-        private static IDictionary<string, string> outboundAlgorithmMap = new Dictionary<string, string>() 
```

Comment:
When using `SecurityAlgorithms.RsaSha256Signature`, the xmldsig name is directly used as-is to build the JWT header:

![image](https://cloud.githubusercontent.com/assets/6998306/11226655/91b21b1e-8d81-11e5-8233-a781828b6875.png)

It's not really an issue when using IdentityModel 5, since it will accept the long name when validating a token, but it crashes on IdentityModel 4, as reported by @jgoelten:

![image](https://cloud.githubusercontent.com/assets/6998306/11226739/17a9fce6-8d82-11e5-9a1d-d679bae444c1.png)

Using `SecurityAlgorithms.RSA_SHA256` is a way to work around this breaking change, but it feels a bit hacky :smile: